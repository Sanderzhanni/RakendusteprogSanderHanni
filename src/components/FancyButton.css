.FancyButton{
    color: black;
}

.FancyButton2{
    color: black;
}