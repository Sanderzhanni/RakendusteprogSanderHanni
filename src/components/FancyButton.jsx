import React from "react";
import PropTypes from "prop-types";
import "./FancyButton.css";

const FancyButton = ({children}) =>(
    <div className={"FancyButton"}>
        <div>{children}</div>
    </div>
);





FancyButton.propTypes = {
  children: PropTypes.string.isRequired
};

export default FancyButton;