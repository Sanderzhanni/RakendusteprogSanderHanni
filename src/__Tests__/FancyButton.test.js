import React from "react";
import renderer from "react-test-renderer";
import FancyButton from "../components/FancyButton.jsx";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

describe("FancyButton component", () =>{
    test("snapshot", () =>{
        const component = renderer.create(
            <FancyButton onClick={() => 0}>
                Hello
            </FancyButton>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

it("callback functionality test", () =>{
    let x = 0;
    const component = shallow(
        <button onClick={() => x++}>add</button>
    );
    component.find("button").simulate("click");
    expect(x).toBe(1);
});