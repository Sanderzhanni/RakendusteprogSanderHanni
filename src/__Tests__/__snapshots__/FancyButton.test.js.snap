// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyButton component snapshot 1`] = `
<div
  className="FancyButton"
>
  <div>
    Hello
  </div>
</div>
`;
